scipy/stats/tests/test_distributions.py

```diff
@@ -1366,6 +1366,9 @@ def test_fit(self, dist, sgn):
 
 
 class TestPareto:
+    def setup_method(self):
+        np.random.seed(1234)
```

Comment:
I would prefer to not do this and explicitly use a `np.random.Generator` RNG where needed. It's even more confusing here since launching a single test from the class would have a different state than launching on the whole class.

Reply:
It was my impression that `setup_method` runs before each method so the state is the same whether the class of tests is run / an individual method, but I can also instantiate a generator and then pass it to `rvs(..., random_state=rng)`, is that what you had in mind?

Reply:
> It was my impression that `setup_method` runs before each method so the state is the same whether the class of tests is run / an individual method,

Indeed, I just checked the [doc](https://docs.pytest.org/en/6.2.x/xunit_setup.html?highlight=setup#method-and-function-level-setup-teardown). But as said there, we should use fixtures instead. This is mostly there to catch people with background with xunit, etc.



> but I can also instantiate a generator and then pass it to `rvs(..., random_state=rng)`, is that what you had in mind?

Yes this is better. Setting aside the fixture vs setup, this is the way to go with NumPy RNG now (using a generator, be it `np.random.RandomState`, only in tests, or `np.random.Generator`, in general), cf NEP19. For new code this is really something we should do. Lot of people are using our tests/examples as default implementation, hence it's important we do things the recommended way.

Reply:
> For new code this is really something we should do. Lot of people are using our tests/examples as default implementation, hence it's important we do things the recommended way.
> 
> I am tempted to propose a fixture which is just declaring a RNG generator so that we just use this in tests.

Happy to follow this practice! For now I kept the fixture's scope class specific, but in the future it could be used in the entire module.

Reply:
Works with me, thanks for the update.